saleor/graphql/meta/types.py

```diff
@@ -15,13 +17,48 @@ class MetadataItem(graphene.ObjectType):
     value = graphene.String(required=True, description="Value of a metadata item.")
 
 
+class Metadata(GenericScalar):
+    """Metadata is a map of key-value pairs, both keys and values are `String`.
+
+    Example:
+    ```
+    {
+        "key1": "value1",
+        "key2": "value2"
+    }
+    ```
+
+    """
+
+
+def _filter_metadata(metadata, keys):
+    if keys is None:
+        return metadata
+    return {key: value for key, value in metadata.items() if key in keys}
+
+
 class ObjectWithMetadata(graphene.Interface):
     private_metadata = NonNullList(
         MetadataItem,
         required=True,
         description=(
-            "List of private metadata items."
-            "Requires proper staff permissions to access."
+            "List of private metadata items. Requires staff permissions to access."
+        ),
+    )
+    private_metafield = graphene.String(
+        args={"key": graphene.NonNull(graphene.String)},
+        description=(
+            "A single key from private metadata. "
+            "Requires staff permissions to access. "
+            "Use GraphQL aliases to fetch multiple keys."
```

Comment:
All new fields should be marked with ADDED_IN_32 label, also should we mark it as preview?

Reply:
Good catch!

Reply:
@maarcingebala I've added the version string to the end of field descriptions because otherwise, the schema becomes a long list of "New in Saleor 3.2...". I've also reworded the "New in Saleor X.Y" to say "Added in Saleor X.Y" because those strings will remain in place until we release 4.0. It's not a feature, so I'm unsure about the preview status. @timuric?

Reply:
I've talked to @timuric about it. We'll keep it marked as Feature Preview until 3.3.